server/apps/immich/src/api-v1/asset/asset-repository.ts

```diff
@@ -287,6 +291,18 @@ export class AssetRepository implements IAssetRepository {
    */
   async update(asset: AssetEntity, dto: UpdateAssetDto): Promise<AssetEntity> {
     asset.isFavorite = dto.isFavorite ?? asset.isFavorite;
+    const tags: TagEntity[] = [];
+
+    if (dto.tagIds && dto.tagIds.length > 0) {
+      for (const tagId of dto.tagIds) {
+        const tag = await this.tagRepository.findOne({ where: { id: tagId } });
```

Comment:
a findAll for an array of tagIds should be more performant and more elegant because it only returns those tags that are found (and not possibly null)